scripts/create_ts_for_integration_tests.py

```diff
@@ -0,0 +1,207 @@
+        "110: hourly values, is_step=True, 1969-10-01 - 1970-03-01, numeric",
+        "111: every minute, is_step=True, 1969-12-31 - 1970-01-02, numeric",
+        "112: every second, is_step=True, 1969-12-31 23:30:00 - 1970-01-01 00:30:00, numeric",
+        "113: every millisecond, is_step=True, 1969-12-31 23:59:58.500 - 1970-01-01 00:00:01.500, numeric",
+        "114: 1mill dps, random distribution, 1950-2020, numeric",
+        "115: 1mill dps, random distribution, 1950-2020, string",
+        "116: 5mill dps, 2k dps (.1s res) burst per day, 2000-01-01 12:00:00 - 2013-09-08 12:03:19.900, numeric",
+    ]
+]
+
+
+def create_dense_rand_dist_ts(xid, seed, n=1_000_000):
+    np.random.seed(seed)
+    idx = np.sort(
+        np.random.randint(
+            pd.Timestamp("1950-01-01").value // int(1e6),
+            pd.Timestamp("2020-01-01").value // int(1e6),
+            n,
+        )
+    )
+    dupe = idx[:-1] == idx[1:]
+    idx[1:][dupe] += 1  # fingers crossed
+    idx = pd.to_datetime(idx, unit="ms").sort_values()
+    assert idx.is_unique, "not unique idx"
+    return pd.DataFrame(
+        {xid: np.arange(n) - 499_999},
+        index=idx,
+    )
+
+
+def create_bursty_ts(xid, offset):
+    idxs = []
+    for day in pd.date_range(start="2000", periods=5000, freq="D"):
+        idxs.append(pd.date_range(start=day + pd.Timedelta("12h"), periods=2000, freq="100ms"))
+    idxs = np.concatenate(idxs)
+    return pd.DataFrame({xid: idxs.astype(np.int64) // int(1e6) + offset}, index=idxs)
+
+
+def delete_all_time_series(ts_api):
+    ts_api.delete(external_id=NAMES, ignore_unknown_ids=True)
+    print(f"Deleted ts: {len(NAMES)=}")
+    time.sleep(10)
+
+
+def create_all_time_series(ts_api):
+    ts_add = (ts_lst := []).append
+    df_add = (df_lst := []).append
+    ts_add(TimeSeries(name=NAMES[0], external_id=NAMES[0], is_string=False, metadata={"offset": 1, "delta": 10}))
+    arr = np.arange(-100, 101, 10)
+    df_add(pd.DataFrame({NAMES[0]: arr + 1}, index=pd.to_datetime(arr, unit="ms")))
+
+    ts_add(TimeSeries(name=NAMES[1], external_id=NAMES[1], is_string=True, metadata={"offset": 2, "delta": 10}))
+    df_add(pd.DataFrame({NAMES[1]: (arr + 2).astype(str)}, index=pd.to_datetime(arr, unit="ms")))
+
+    weekly_idx = pd.date_range(start="1950", end="2000", freq="1w")
+    arr = weekly_idx.to_numpy("datetime64[ms]").astype(np.int64)
+    for i, name in enumerate(NAMES[2:53], 3):
+        ts_add(
+            TimeSeries(name=name, external_id=name, is_string=False, metadata={"offset": i, "delta": UNIT_IN_MS["w"]})
+        )
+        df_add(pd.DataFrame({name: arr + i}, index=weekly_idx))
+
+    for i, name in enumerate(NAMES[53:103], i + 1):
+        ts_add(
+            TimeSeries(name=name, external_id=name, is_string=True, metadata={"offset": i, "delta": UNIT_IN_MS["w"]})
+        )
+        df_add(pd.DataFrame({name: (arr + i).astype(str)}, index=weekly_idx))
+
+    i = 103
+    for is_step in [False, True]:
```

Comment:
Question: Would you get a noticeable speedup if you stored all the "i + 1", "i + 2", "i + 3", "i + 4", and "i + 5" variables that get computed repeatedly in this loop? I.e., set variables like `t_1 = i + 1`, `t_2 = i + 2`, etc and use those instead of the repeated calculations. 🤔

Reply:
This is a script that - if no one touches the Cognite project for the "Python-SDK-integration-tests" - never has to be run again 😉

Reply:
Assume that someone will delete all your test data tomorrow. Ideally, you have a session-scoped fixture which validates the data you need and reingests it if any is missing.

Reply:
That would not work; it wouldn’t be possible to know if the underlying data was wrong or the code that fetched it. I should probably add a note about that reingesting should always be done by a previous/released version